pkg/cmd/ssh/ssh.go

```diff
@@ -43,5 +45,74 @@ func NewCmdSSH(f util.Factory, o *SSHOptions) *cobra.Command {
 	cmd.Flags().DurationVar(&o.WaitTimeout, "wait-timeout", o.WaitTimeout, "Maximum duration to wait for the bastion to become available.")
 	cmd.Flags().BoolVar(&o.KeepBastion, "keep-bastion", o.KeepBastion, "Do not delete immediately when gardenctl exits (Bastions will be garbage-collected after some time)")
 
+	registerCompletionFuncForFlags(cmd, f, o.IOStreams)
+
 	return cmd
 }
+
+func registerCompletionFuncForFlags(cmd *cobra.Command, f util.Factory, ioStreams util.IOStreams) {
+	utilruntime.Must(cmd.RegisterFlagCompletionFunc("cidr", completionWrapper(f, ioStreams, cidrFlagCompletionFunc)))
+}
+
+type cobraCompletionFunc func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
+type cobraCompletionFuncWithError func(f util.Factory) ([]string, error)
+
+func completionWrapper(f util.Factory, ioStreams util.IOStreams, completer cobraCompletionFuncWithError) cobraCompletionFunc {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		result, err := completer(f)
+		if err != nil {
+			fmt.Fprintf(ioStreams.ErrOut, "%v\n", err)
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		return util.FilterStringsByPrefix(toComplete, result), cobra.ShellCompDirectiveNoFileComp
+	}
+}
+
+func cidrFlagCompletionFunc(f util.Factory) ([]string, error) {
+	var addresses []string
+
+	ctx := f.Context()
+
+	publicIPs, err := f.PublicIPs(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ip := range publicIPs {
+		cidr := ipToCIDR(ip)
+		addresses = append(addresses, fmt.Sprintf("%s\t<public>", cidr))
+	}
+
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	includeFlags := net.FlagUp
+	excludeFlags := net.FlagLoopback
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return nil, err
+		}
+
+		if is(iface, includeFlags) && isNot(iface, excludeFlags) {
```

Comment:
What is happening here? Why this bit operation?

Reply:
I want to trim down the list of suggestions and only include interfaces that are `up` and exlude loopback interfaces.
Or what do you want to know?
See also https://cs.opensource.google/go/go/+/master:src/net/interface.go;l=36-73?q=FlagBroadcast&ss=go%2Fgo